spec/features/visitors/confirm_email_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+
+feature 'Confirm email' do
+  VALID_PASSWORD = 'Val!d Pass w0rd'.freeze
+
+  scenario 'confirms valid email and sets valid password' do
+    sign_up_with('test@example.com')
+    confirm_last_user
+
+    expect(page).to have_content t('devise.confirmations.confirmed_but_must_set_password')
+    expect(page).to have_title t('titles.confirmations.show')
+    expect(page).to have_content t('forms.confirmation.show_hdr')
+
+    fill_in 'password_form_password', with: VALID_PASSWORD
+    click_button t('forms.buttons.submit.default')
+
+    expect(current_url).to eq phone_setup_url
+    expect(page).to_not have_content t('devise.confirmations.confirmed_but_must_set_password')
+  end
+
+  scenario 'confirms valid email and submits invalid password', js: true do
+    sign_up_with('test@example.com')
+    confirm_last_user
+
+    expect(page).to have_content t('devise.confirmations.confirmed_but_must_set_password')
+    expect(page).to have_title t('titles.confirmations.show')
+    expect(page).to have_content t('forms.confirmation.show_hdr')
+
+    fill_in 'password_form_password', with: 'password'
+    click_button t('forms.buttons.submit.default')
+    fill_in 'password_form_password', with: VALID_PASSWORD
+    click_button t('forms.buttons.submit.default')
```

Comment:
NOTE: this test did NOT fail before I made the fix for this issue by removing `this.form.addEventListener('change', e => validateField(e.target));`. I noticed when testing locally that clicking "enter" button did not have this bug, only trying to click the button itself. So maybe capybara is doing something similar, which is why I could not reproduce the bug in a test env?